apps/labs/posts/uarray-update-api-changes-overhead-and-comparison-to-__array_function__.md

```diff
@@ -0,0 +1,703 @@
+                        return NotImplemented
+                    converted.append(da.asarray(d.value))
+                except (TypeError, ValueError):
+                    return NotImplemented
+            else:
+                # Pass through everything else, we only want to dispatch on arrays.
+                converted.append(d.value)
+        return converted
+```
+
+``` python
+with ua.set_backend(DaskBackend):
+    print(ones((3, 4)))
+```
+
+    dask.array<ones, shape=(3, 4), dtype=float64, chunksize=(3, 4)>
+
+Note how a simple context manager allows things to magically work,
+unlike in `__array_function__` where array creation functions are not
+supported at all.
+
+Here are a few things that could be done by Dask:
+
+1.  It could depend on `unumpy` and become a true, agnostic, meta-array.
+2.  It could define a backend factory, which automatically inserts the
+    chunk sizes into functions.
+
+Next, let's move on to XArray.
+
+``` python
+import xarray
+
+def xarray_ones(shape, dtype=None):
+    data = ones(tuple(shape.values()), dtype=dtype)
+    return xarray.DataArray(data, dims=tuple(shape.keys()))
+
+class XarrayBackend:  # Ideally, this would be the xarray module itself.
+    __ua_domain__ = "numpy"
+    
+    @staticmethod
+    def __ua_function__(f, a, kw):
+        with ua.skip_backend(XarrayBackend):
+            if f is ones:
+                return xarray_ones(*a, **kw)
+        
+        return NotImplemented
+    
+    @staticmethod
+    def __ua_convert__(dispatchables, coerce):
+        converted = []
+        
+        for d in dispatchables:
+            if d.type is np.ndarray:
+                try:
+                    # We ONLY want to coerce if coercion is true.
+                    if not hasattr(d.value, '__array_interface__') and not (coerce and d.coercible):
+                        return NotImplemented
+                    converted.append(xarray.Dataset(d.value))
+                except (TypeError, ValueError):
+                    return NotImplemented
+            elif d.type is shape_type:
+                if not (isinstance(d.value, dict)
+                        and all(
+                            isinstance(k, str)
+                            and isinstance(v, numbers.Integral)
+                            for k, v in d.value.items()
+                        )):
+                    return NotImplemented
+
+                converted.append({k: int(v) for k, v in d.value.items()})
+            elif d.type is np.dtype:
+                try:
+                    converted.append(np.dtype(d.value))
+                except (TypeError, ValueError):
+                    return NotImplemented
+            else:
+                # Handle the base case, pass through everything else
+                converted.append(d.value)
+        return converted
+```
+
+``` python
+with ua.set_backend(XarrayBackend):
+    print(ones({'t': 3, 'x': 5}))
+```
+
+    <xarray.DataArray (t: 3, x: 5)>
```

Comment:
I'm getting the Vercel build error: `error [Error: [next-mdx-remote] error compiling MDX:`

Full error message:
```
Unexpected character `(` (U+0028) after member name,
expected a character that can start an attribute name, such as
a letter, `$`, or `_`; whitespace before attributes; or the end of the tag

  655 | ```
  656 |
> 657 |     <xarray.DataArray (t: 3, x: 5)>
      |                       ^
  658 |     array([[1., 1., 1., 1., 1.],
  659 |            [1., 1., 1., 1., 1.],
```

[Link to error](https://vercel.com/quansight/labs/CxQa2nRa1B9XDVDyQVEtDUEYeb3Q)

The context is showing the result of some code. [Link to this section in the old blog website.](https://labs.quansight.org/blog/2019/07/uarray-update-api-changes-overhead-and-comparison-to-__array_function__#:~:text=%3C-,xarray.,-DataArray%20)

Similar problem to #647: output of code blocks formatted as plain text, not in code blocks, causing problems to Vercel